dfinity/nns-dapp
# Chore: Remove unused add_pending_transaction functionality

# Motivation

Remove unused functionality.

The method `add_pending_notify_swap` was created for SNS-1 before the ticketing system was in place. It helped with the edge case of a user making the transaction and then not notifying the swap canister about it.

This flow is not necessary anymore with the ticket system.

In this PR, I removed the method `add_pending_notify_swap` and everything related to it.

# Changes

* Remove `addPendingNotifySwap` from NnsDappCanister class.
* Remove the method and types from NNS Dapp candid and related files.
* Remove unused `NotAuthorizedError` which was used only in `addPendingNotifySwap`.
* Remove method `add_pending_transaction` and related methods from backend AccountsStore.
* Remove canister module swap that was used only to handle swap transactions.
* Remove candid method `add_pending_notify_swap` from `main.rs` and related method.
* Remove handling the ParticipateSwap transaction in the periodic task runner.
* Remove unused time_millis.

# Tests

I deployed the canister locally and tested all the different transactions. All worked as expected.

* Remove tests for `addPendingNotifySwap`.
* Remove method from nns-dapp-exports tests.
* Remove backend tests related to the functionality.

# Todos
